src/Adapter/Carrier/CommandHandler/SetCarrierRangesHandler.php

```diff
@@ -0,0 +1,76 @@
+<?php
+/**
+ * Copyright since 2007 PrestaShop SA and Contributors
+ * PrestaShop is an International Registered Trademark & Property of PrestaShop SA
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.md.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to https://devdocs.prestashop.com/ for more information.
+ *
+ * @author    PrestaShop SA and Contributors <contact@prestashop.com>
+ * @copyright Since 2007 PrestaShop SA and Contributors
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ */
+
+namespace PrestaShop\PrestaShop\Adapter\Carrier\CommandHandler;
+
+use Carrier;
+use PrestaShop\PrestaShop\Adapter\Carrier\Repository\CarrierRangeRepository;
+use PrestaShop\PrestaShop\Adapter\Carrier\Repository\CarrierRepository;
+use PrestaShop\PrestaShop\Adapter\Carrier\Validate\CarrierRangesValidator;
+use PrestaShop\PrestaShop\Core\CommandBus\Attributes\AsCommandHandler;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\Command\SetCarrierRangesCommand;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\CommandHandler\SetCarrierRangesHandlerInterface;
+use PrestaShop\PrestaShop\Core\Domain\Carrier\ValueObject\CarrierId;
+
+/**
+ * Handles query which gets carrier range
+ */
+#[AsCommandHandler]
+final class SetCarrierRangesHandler implements SetCarrierRangesHandlerInterface
+{
+    public function __construct(
+        private readonly CarrierRepository $carrierRepository,
+        private readonly CarrierRangeRepository $carrierRangeRepository,
+        private readonly CarrierRangesValidator $rangesValidator,
+    ) {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handle(SetCarrierRangesCommand $command): CarrierId
+    {
+        // Before all, we need to validate ranges
+        $this->rangesValidator->validate($command->getRanges());
+
+        // Switch calculation method if needed
+        $carrier = new Carrier();
```

Comment:
Why a new carrier? Why not using `$this->carrierRepository->get($command->getCarrierId())`?

Reply:
It's because our duplication carrier system, that will change in https://github.com/PrestaShop/PrestaShop/issues/36263 :)

Reply:
Beware @boherm, that duplication can be avoided only if the carrier is not yet in an order.

If it's already used in an order, we must duplicate the carrier to retain the original data.

Reply:
Yes, totally @Hlavtox, that's I have in mind!